api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java

```diff
@@ -985,4 +985,12 @@ public DbUser confirmProfile() {
     dbUser.setProfileLastConfirmedTime(new Timestamp(clock.instant().toEpochMilli()));
     return userDao.save(dbUser);
   }
+
+  /** Confirm that a user has either reported any AoU-related publications, or has none. */
+  @Override
+  public DbUser confirmPublications() {
```

Comment:
I think that is a good principle in general.  Here, I'm not so sure because it's important that only the user making the API request has can do this - we are recording a specific user action as a compliance signal.

Having said that ... I should probably add an audit event here as well.

Reply:
updated to use our standard access module updating process